metrics/src/metrics.rs

```diff
@@ -135,11 +145,39 @@ impl MetricsAgent {
             }
 
             let now = Instant::now();
-            if now.duration_since(last_write_time) >= write_frequency && !points.is_empty() {
-                debug!("run: writing {} points", points.len());
+            if now.duration_since(last_write_time) >= write_frequency_secs && !points.is_empty() {
+                let num_points = points.len();
+                let points_written;
+                debug!("run: attempting to write {} points", points.len());
+                if points.len() > max_points {
+                    warn!(
+                        "max submission rate of {} datapoints per second exceeded.  only the
+                    first {} of {} points will be submitted",
+                        max_points_per_sec,
+                        max_points,
+                        points.len()
+                    );
+                    points.truncate(max_points);
+                }
+                points_written = points.len();
                 writer.write(points);
+
                 points = Vec::new();
                 last_write_time = now;
+
+                submit(
```

Comment:
one last nit: rather than calling `submit()`, we're already in the right place to call `writer.write()` so just append this new dataport to the `points` vec above and it'll get automatically included in the `writer.write(points);`!

Reply:
Even if we are at the max rate and that one final append means we will exceed our rate limit by 1?

Reply:
Yeah.  This is one of the datapoints that we really want